src/token_request_handler.ts

```diff
@@ -46,6 +51,24 @@ export class BaseTokenRequestHandler implements TokenRequestHandler {
     return (response as TokenErrorJson).error === undefined;
   }
 
+  performRevokeTokenRequest(
+      configuration: AuthorizationServiceConfiguration,
+      request: RevokeTokenRequest): Promise<boolean> {
+    let revokeTokenResponse = this.requestor.xhr<any>({
+      url: configuration.revocationEndpoint + '?' + this.utils.stringify(request.toStringMap()),
+      dataType: 'json',  // adding implicit dataType
+      method: 'GET'
```

Comment:
You should be able to reject the promise when an `xhr` fails with the appropriate `AppAuthError` with the underlying HTTP error code. If `xhr` does not do that already, it should.